Gems/ExternalProfilers/TracyProfiler/Code/CMakeLists.txt

```diff
@@ -0,0 +1,81 @@
+#
+# Copyright (c) Contributors to the Open 3D Engine Project.
+# For complete copyright and license terms please see the LICENSE at the root of this distribution.
+#
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+#
+#
+
+# Add Tracy dependency using FetchContent
+include(FetchContent)
+FetchContent_Declare(
+    Tracy
+    GIT_REPOSITORY "https://github.com/wolfpld/tracy"
+    GIT_TAG "5d542dc09f3d9378d005092a4ad446bd405f819a" # version 0.11.1
+)
+
+set(TRACY_ENABLE ON)
+set(TRACY_CALLSTACK ON)
+set(TRACY_ON_DEMAND ON)
+ly_append_configurations_options(
+    DEFINES
+        TRACY_ENABLE
```

Comment:
Isn't this super dangerous, like, doesn't ly_append_configurations_options set this option on for every single compilation unit in the entire engine, which requires a full recompile of the entire engine?

If so that would make it impossible to ever have this module included with an install or prebuilt since turning it on requires a full rebuild of the engine.

But I don't see anywhere in the o3de engine code that uses this define?  where does it actually need to be  defined (what targets?)  You can always add it as target_compile_options() to specific targets, after fetchcontent_makeavailable

Reply:
Yeah it was by purpose but wasn't doing anything at the end. Its the Tracy documentation which outlines to define that for every files and as sampling wasn't working on windows I thought it might be related, turns out it is not so I removed it altogether